plasmapy/formulary/mathematics.py

```diff
@@ -88,3 +88,77 @@ def Fermi_integral(
         return integral_arr
     else:
         raise TypeError(f"Improper type {type(x)} given for argument x.")
+
+
+def rot_a_to_b(a: np.ndarray, b: np.ndarray) -> np.ndarray:
+    r"""
+    Calculates the 3D rotation matrix that will rotate vector ``a`` to be aligned
+    with vector ``b``. The rotation matrix is calculated as follows. Let
+
+    .. math::
+        \vec v = \vec a \times \vec b
+
+    and let :math:`\theta` be the angle between :math:`\vec a`
+    and :math:`\vec b` such that the projection of :math:`\vec a` along
+    :math:`\vec b` is
+
+    .. math::
+        c = \vec a \cdot \vec b \cos\theta
+
+    Then the rotation matrix :math:`R` is
+
+    .. math::
+        R = I + v_x + v_x^2 \frac{1}{1 + c}
+
+
+    where :math:`I` is the identity matrix and :math:`v_x` is the
+    skew-symmetric cross-product matrix of :math:`v` defined as
+
+    .. math::
+        v_x = \begin{bmatrix}
+                0 & -v_3 & v_2 \\
+                v_3 & 0 & -v_1 \\
+                -v_2 & v_1 & 0
+            \end{bmatrix}
+
+    Note that this algorithm fails when :math:`1+c=0`, which occurs when :math:`a` and
+    :math:`b` are anti-parallel. However, since the correct rotation matrix
+    in this case is simply :math:`R=-I`, this function just handles this
+    special case explicitly.
+
+    This algorithm is based on `this discussion <https://math.stackexchange.com/questions/180418/calculate-rotation-matrix-to-align-vector-a-to-vector-b-in-3d/476311#476311>`_ on StackExchange:
```

Comment:
```suggestion
    The algorithm is based on
    `this discussion <https://math.stackexchange.com/a/476311>`_ on StackExchange.
```
Use the shorter link so the docstrings stay within our max line width.